src/ray/common/task/task.h

```diff
@@ -64,6 +75,34 @@ class Task {
 
   std::string DebugString() const;
 
+  /// Whether this task has at least one coscheduled tasks.
+  bool IsVectorTask() const {
+    return task_spec_vector_ != nullptr && task_spec_vector_->size() > 1;
+  }
+
+  /// Return the list of coscheduled tasks, including the base task spec.
+  const std::vector<TaskSpecification> &GetTaskSpecificationVector() const {
+    RAY_CHECK(task_spec_vector_ != nullptr);
+    return *task_spec_vector_;
+  }
+
+  /// Remove the given task ids from the task vector.
+  std::vector<TaskSpecification> RemoveTaskSpecsFromVector(std::vector<TaskID> task_ids) {
+    RAY_CHECK(task_spec_vector_ != nullptr);
+    std::vector<TaskSpecification> out;
+    auto it = task_spec_vector_->begin();
+    while (it != task_spec_vector_->end()) {
+      if (std::count(task_ids.begin(), task_ids.end(), it->TaskId()) > 0) {
+        out.push_back(*it);
+        task_spec_vector_->erase(it);
```

Comment:
Do you need to do `it = task_spec_vector_->erase(it);` here? This might also work but technically `it` gets invalidated AFAIK.

Reply:
I believe this is specifically the right way to delete from a list as you iterate through it.

Reply:
I looked at https://en.cppreference.com/w/cpp/container/vector/erase, where it says
`Invalidates iterators and references at or after the point of the erase, including the end() iterator.`.

Reply:
Yeah but not itself.

https://www.techiedelight.com/remove-elements-vector-inside-loop-cpp/

Reply:
I definitely don't want to belabor the point, but doesn't the link say you should do `it = task_spec_vector_->erase(it);` (in option 1)? :D

Reply:
oh that's what you meant... yeah makes sense. Not sure why it worked.

Reply:
makes sense, it's probably one of these things that are officially undefined but work in actual implementations